src/Toolkit/Toolkit/UI/Controls/BasemapGallery/BasemapGalleryItem.cs

```diff
@@ -246,9 +246,54 @@ internal bool EqualsBasemap(Basemap? other)
                 return false;
             }
 
-            return other == Basemap || other.Name == Basemap?.Name
-                                    || (other.Item?.ItemId != null && other.Item?.ItemId == Basemap?.Item?.ItemId)
-                                    || (other.Uri != null && other.Uri == Basemap?.Uri);
+            return other == Basemap || (other.Item?.ItemId != null && other.Item?.ItemId == Basemap?.Item?.ItemId)
+                                    || (other.Uri != null && other.Uri == Basemap?.Uri)
+                                    || AreBasemapsEqualByLayers(Basemap, other);
+        }
+
+        private static bool AreBasemapsEqualByLayers(Basemap? basemap1, Basemap? basemap2)
+        {
+            if (basemap1 == null || basemap2 == null) return false;
+
+            return LayersEqual(basemap1.BaseLayers, basemap2.BaseLayers)
+                && LayersEqual(basemap1.ReferenceLayers, basemap2.ReferenceLayers);
+        }
+
+        private static bool LayersEqual(LayerCollection layers1, LayerCollection layers2)
+        {
+            return layers1.Count == layers2.Count
+                && layers1.Zip(layers2, (layer1, layer2) => new { Layer1 = layer1, Layer2 = layer2 })
+                    .All(pair => LayerEquals(pair.Layer1, pair.Layer2));
+        }
```

Comment:
Optional:
You can pass LayersEquals directly to Zip, then just check that all returned bools are true.

```cs
bool LayersEqual(LayerCollection layers1, LayerCollection layers2)
{
    return layers1.Count == layers2.Count 
        && layers1.Zip(layers2, LayerEquals).All(equal => equal);
}
```

Reply:
```suggestion
        private static bool LayersEqual(LayerCollection layers1, LayerCollection layers2)
        {
            return layers1.Count == layers2.Count
                && layers1.Zip(layers2, LayerEquals).All(equal => equal);
        }
```